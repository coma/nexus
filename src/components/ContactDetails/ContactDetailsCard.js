import React from 'react';
import PropTypes from 'prop-types';

import {
  Wrapper,
  Picture,
  Name,
  Details,
  Email,
  Phone,
} from './ContactDetailsCard.style';

function ContactDetailsCard({ contact }) {
  return (
    <Wrapper>
      <Picture sizes={contact.picture} />
      <Name>
        {contact.name.title} 
        {' '}
        {contact.name.first} 
        {' '}
        {contact.name.last}
      </Name>
      <Details>
        <Email href={`mailto:${contact.email}`}>{contact.email}</Email>
        <Phone href={`tel:${contact.phone}`}>{contact.phone}</Phone>
      </Details>
    </Wrapper>
  );
}

ContactDetailsCard.propTypes = {
  contact: PropTypes.shape({
    name: PropTypes.shape({
      title: PropTypes.string,
      first: PropTypes.string,
      last: PropTypes.string,
    }).isRequired,
    picture: PropTypes.shape({
      large: PropTypes.string,
      medium: PropTypes.string,
    }).isRequired,
    email: PropTypes.string,
    phone: PropTypes.string,
  }).isRequired,
};

export default ContactDetailsCard;
