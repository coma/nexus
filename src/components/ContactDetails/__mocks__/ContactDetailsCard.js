export default ({ contact }) => contact.id;
