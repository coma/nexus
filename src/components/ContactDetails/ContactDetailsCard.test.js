import React from 'react';
import { render } from 'react-testing-library';

import Theme from '../Theme';
import ContactDetailsCard from './ContactDetailsCard';

describe('The ContactDetailsCard component', () => {
  it('should render properly', () => {
    const { container } = render(
      <Theme>
        <ContactDetailsCard
          contact={{
            name: {
              title: 'Dr.',
              first: 'Strange',
              last: 'Love',
            },
            picture: {
              large: 'large.jpg',
              medium: 'medium.jpg',
            },
            email: 'love@the.bomb',
            phone: '12345678',
          }}
        />
      </Theme>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
