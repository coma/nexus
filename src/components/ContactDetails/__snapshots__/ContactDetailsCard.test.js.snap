// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ContactDetailsCard component should render properly 1`] = `
<div
  class="sc-bdVaJa fkcOCo"
>
  <div
    class="sc-bZQynM fHwPtR"
    sizes="[object Object]"
  />
  <div
    class="sc-htpNat gUfSqx"
  >
    Dr.
     
    Strange
     
    Love
  </div>
  <div
    class="sc-bxivhb dJdFkY"
  >
    <a
      class="sc-ifAKCX kKPrLR"
      href="mailto:love@the.bomb"
    >
      love@the.bomb
    </a>
    <a
      class="sc-EHOje jvHvlZ"
      href="tel:12345678"
    >
      12345678
    </a>
  </div>
</div>
`;
